Fix update counters of multiple records with touch: true

kamipo opened the pull request:
Currently does not work the following example in the doc:

```ruby
  # For the Posts with id of 10 and 15, increment the comment_count by 1
  # and update the updated_at value for each counter.
  Post.update_counters [10, 15], comment_count: 1, touch: true
  # Executes the following SQL:
  # UPDATE posts
  #    SET comment_count = COALESCE(comment_count, 0) + 1,
  #    `updated_at` = '2016-10-13T09:59:23-05:00'
  #  WHERE id IN (10, 15)
```

kamipo committed:
Fix update counters of multiple records with touch: true

Currently does not work the following example in the doc:

```ruby
  # For the Posts with id of 10 and 15, increment the comment_count by 1
  # and update the updated_at value for each counter.
  Post.update_counters [10, 15], comment_count: 1, touch: true
  # Executes the following SQL:
  # UPDATE posts
  #    SET comment_count = COALESCE(comment_count, 0) + 1,
  #    `updated_at` = '2016-10-13T09:59:23-05:00'
  #  WHERE id IN (10, 15)
```

pixeltrix commented:
Why do we need the find? Surely there must be a way to build a hash of updates without instantiating an object. We also seem to be finding an object so that we can call `current_time_from_proper_timezone` on it which itself calls `self.class.default_timezone` - this all seems a bit convoluted.

kamipo committed:
Push `current_time_from_proper_timezone` and timestamp attributes methods up to class method

Actually these methods don't need instantiation.

kamipo committed:
Counter cache touching don't need object finding anymore

`current_time_from_proper_timezone` and timestamp attributes methods was
pushed up to class method.

kamipo commented:
Good point. I fixed to build a hash of updates without instantiating an object.

kaspth commented:
@kamipo way cool! Good points @pixeltrix 🤘
